assets/dev/js/editor/atomic-widgets/views/a-container.js

```diff
@@ -0,0 +1,347 @@
+import ContainerHelper from 'elementor-editor-utils/container-helper';
+
+const BaseElementView = require( 'elementor-elements/views/base' );
+const AContainerView = BaseElementView.extend( {
+	template: Marionette.TemplateCache.get( '#tmpl-elementor-a-container-content' ),
+
+	emptyView: AEmptyView,
+
+	tagName() {
+		return this.model.getSetting( 'tag' ) || 'div';
+	},
+
+	getChildViewContainer() {
+		this.childViewContainer = '';
+
+		return Marionette.CompositeView.prototype.getChildViewContainer.apply( this, arguments );
+	},
+
+	className() {
+		return `${ BaseElementView.prototype.className.apply( this ) } e-con a-con`;
+	},
+
+	// TODO: Copied from `views/column.js`.
+	ui() {
+		var ui = BaseElementView.prototype.ui.apply( this, arguments );
+
+		ui.percentsTooltip = '> .elementor-element-overlay .elementor-column-percents-tooltip';
+
+		return ui;
+	},
+
+	// TODO: Copied from `views/column.js`.
+	attachElContent() {
+		BaseElementView.prototype.attachElContent.apply( this, arguments );
+
+		const $tooltip = jQuery( '<div>', {
+			class: 'elementor-column-percents-tooltip',
+			'data-side': elementorCommon.config.isRTL ? 'right' : 'left',
+		} );
+
+		this.$el.children( '.elementor-element-overlay' ).append( $tooltip );
+	},
+
+	// TODO: Copied from `views/column.js`.
+	getPercentSize( size ) {
+		if ( ! size ) {
+			size = this.el.getBoundingClientRect().width;
+		}
+
+		return +( size / this.$el.parent().width() * 100 ).toFixed( 3 );
+	},
+
+	// TODO: Copied from `views/column.js`.
+	getPercentsForDisplay() {
+		const width = +this.model.getSetting( 'width' ) || this.getPercentSize();
+
+		return width.toFixed( 1 ) + '%';
+	},
+
+	onResizeStart() {
+		if ( this.ui.percentsTooltip ) {
+			this.ui.percentsTooltip.show();
+		}
+	},
+
+	onResize() {
+		// TODO: Copied from `views/column.js`.
+		if ( this.ui.percentsTooltip ) {
+			this.ui.percentsTooltip.text( this.getPercentsForDisplay() );
+		}
+	},
+
+	onResizeStop() {
+		if ( this.ui.percentsTooltip ) {
+			this.ui.percentsTooltip.hide();
+		}
+	},
+
+	renderOnChange() {
+		BaseElementView.prototype.renderOnChange.apply( this, arguments );
+
+		const classes = this.getContainer().settings.get( 'classes' );
+		if ( classes && classes.value && classes.value[ 0 ] ) {
+			this.$el.addClass( classes.value[ 0 ] );
+		}
+	},
+
+	onRender() {
+		BaseElementView.prototype.onRender.apply( this, arguments );
+
+		// Defer to wait for everything to render.
+		setTimeout( () => {
+			this.droppableInitialize();
+		} );
+	},
+
+	droppableInitialize() {
+		this.$el.html5Droppable( this.getDroppableOptions() );
+	},
+
+	isDroppingAllowed() {
+		return true;
+	},
+
+	filterSettings( newItem ) {
+		if ( 'container' !== newItem?.elType ) {
+			return;
+		}
+
+		const parentContainer = this;
+
+		if ( parentContainer.isBoxedWidth() ) {
+			newItem.settings.content_width = 'full';
+		} else if ( 0 !== parentContainer.getNestingLevel() ) {
+			newItem.settings.content_width = 'full';
+		}
+	},
```

Comment:
I don't think this is necessary. Maybe double check this file to make sure there is no unnecessary methods here.

Reply:
Removed